File: 128-longest-consecutive-sequence/128-longest-consecutive-sequence.cs
public class Solution {
    public int LongestConsecutive(int[] nums) {
        HashSet<int> hs = new HashSet<int>(nums);
        int streak = 0;
        foreach (int x in hs) {
            Console.WriteLine("X is " + x);
            if (!hs.Contains(x - 1)) {
                //start of the streak
                int upper = x;
                int count = 1;
                while(hs.Contains(upper + 1)) {
                    upper += 1;
                    count += 1;
                }
                streak = Math.Max(streak, count);
            }
        }
        return streak;
    }
}

File: 128-longest-consecutive-sequence/128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        s = set(nums)
        streak = 1
        ms = 0
        for x in s:
            print('current', x)
            if x - 1 not in s:
                streak = 1
                cx = x
                while cx + 1 in s:
                    cx += 1
                    streak += 1
                    ms = max(ms, streak)
                ms = max(ms, streak)
        return ms